mlibrary/pride
# [LLIBSEARCH-518] Install and setup `esbuild`

# Overview
[Gulp](https://www.npmjs.com/package/gulp) is Pride's current JavaScript bundler, which is a package that has not been updated in over four years. [ESBuild](https://esbuild.github.io/) has become the preferred JavaScript bundler for LIT projects. To make Pride more up to date and smaller in size, `esbuild` replaces `gulp` and all related dependencies. The `dependency_updates_template.md` pull request template has been updated with the new list of dependencies.

This pull request closes [LIBSEARCH-518](https://mlit.atlassian.net/browse/LIBSEARCH-518).

## `concat.js`
`./concat.js` was created to use NodeJS to concatenate all JavaScript files into one. The generated file is then use for `esbuild` as the package does not have the option to concatenate files.

## Testing
- Install the new list of dependencies (`npm install`).
- Build the repository (`npm run build`).
- Open `./SpecRunner.html` to see if the tests still pass.
- Go to your local Search repository and test the newly generated files in Search:
  ```bash
  rm -rf node_modules/pride
  mkdir node_modules/pride
  cp -r ../pride/* node_modules/pride/.
  npm start
   ``` 
  - Look around [the site](http://localhost:3000/everything) and check to see if there are any console errors, and everything still works as expected.
